### PR TITLE
wxCollapsePane bug  and StyledTextCtrl for python

### PR DESCRIPTION
**Pull request by JacoboSpain**
-> Add StyledTextCtrl for python.
-> BUG generate code for PT_WXPARENT_CP, missing self

### CODE REVIEW

File: output/plugins/additional/xml/additional.pythoncode [hunk 1]
```diff
@@ -269,9 +269,64 @@ Python code generation written by
...
+			#nl self.$name.MarkerSetBackground( wx.stc.STC_MARKNUM_FOLDEREND, wx.BLACK )
+			#nl self.$name.MarkerSetForeground( wx.stc.STC_MARKNUM_FOLDEREND, wx.WHITE )
+			#nl self.$name.MarkerDefine( wx.stc.STC_MARKNUM_FOLDEROPENMID, wx.stc.STC_MARK_BOXMINUS )
+			#nl self.$name.MarkerSetBackground( wx.stc.STC_MARKNUM_FOLDEROPENMID, wx.BLACK)
+			#nl self.$name.MarkerSetForeground( wx.stc.STC_MARKNUM_FOLDEROPENMID, wx.WHITE)
+			#nl self.$name.MarkerDefine( wx.stc.STC_MARKNUM_FOLDERMIDTAIL, wx.stc.STC_MARK_EMPTY )
+			#nl self.$name.MarkerDefine( wx.stc.STC_MARKNUM_FOLDERTAIL, wx.stc.STC_MARK_EMPTY )
+			#nl self.$name.SetSelBackground( True, wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHT ) )
+			#nl self.$name.SetSelForeground( True, wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHTTEXT ) )
+			<!-- @# WARNING: wxPython code generation isn't supported for this widget yet. #nl
```
Comment (sodevel) [inline output/plugins/additional/xml/additional.pythoncode]: You implemented code generation for this control, no need to keep the old placeholder code, just remove it

File: src/codegen/pythoncg.cpp [hunk 2]
```diff
@@ -24,6 +24,9 @@
 // Python code generation writen by
 //   Michal Bližňak - michal.bliznak@gmail.com
 //
+//   Jacobo Costas Costas
```
Comment (sodevel) [inline src/codegen/pythoncg.cpp]: Don't add credits for small fixes or additions. So far these credits have only been used on large additions like support for a new generator language.

File: src/codegen/pythoncg.cpp [hunk 3]
```diff
@@ -102,7 +105,7 @@ wxString PythonTemplateParser::ValueToCode( PropertyType type, wxString value )
 		}
 	case PT_WXPARENT_CP:
 	{
-		result = value + wxT(".GetPane()");
+		result = wxT("self.")  + value + wxT(".GetPane()");
```
Comment (sodevel) [inline src/codegen/pythoncg.cpp]: What are you trying to fix with this? I don't think this is correct, i did a quick test with the current version and the `self.` gets added in the proper places without your change, with your change this should result in `self.self.` in the places.

Comment (JacoboSpain) [reply src/codegen/pythoncg.cpp]: The problem is when elements are inserted into the collapsePane, and it is necessary to call GetPane ()

For example: With original trunk a have a frame:
Frame

The program generate a code:
fail_collapsePane

With my coding:

fail_collapsePane

Insert .self when GetPane is called

Comment (sodevel) [reply src/codegen/pythoncg.cpp]: I still can't reproduce your issue, i have rebuild your layout and this is the generated code:

collapse

This is made by wxFB 48f2f0f, which revision are you using?

Comment (JacoboSpain) [reply src/codegen/pythoncg.cpp]: last revision 0e591aaa7a98ca756359bd5dd8cd2cfa9a7d1140
Compile in Windows 10 64 bits with MSYS32 with gcc 9.1.0

Comment (sodevel) [reply src/codegen/pythoncg.cpp]: Ah i see, thats the reason, my revision is too old, it doesn't contain the changes to the code generation that have been made by myself 😄. I can reproduce the issue now and unfortunately not only Python is affected but also PHP and Lua, but for some reason in Lua the error doesn't manifest. In contrast to the wxStaticBoxSizer case the path to the parent wx element needs to get prepended.

This fix is kind of unrelated to the added control, can you please remove it from this PR and file it as a separate one, plus maybe fixing the other two languages as well?

Comment (JacoboSpain) [reply src/codegen/pythoncg.cpp]: ok

**Pull request closed without merging**